build.gradle

```
@@ -0,0 +1,148 @@
+import org.gradle.crypto.checksum.Checksum
+
+plugins {
+    id 'java'
+    id 'maven-publish'
+    id 'java-library-distribution'
+    id 'org.gradle.crypto.checksum' version '1.1.0'
+}
+
+repositories {
+   mavenCentral()
+}
+
+sourceSets {
+    test {
+        java.srcDir file('src/test/java/')
+        resources.srcDir 'src/test/resources'
+    }
+}
+
+dependencies {
+    implementation 'org.apache.httpcomponents:httpclient:4.5.10'
+    implementation 'com.fasterxml.jackson.core:jackson-databind:2.10.1'
+    implementation 'org.apache.logging.log4j:log4j-api:2.13.2'
+    implementation 'org.apache.logging.log4j:log4j-core:2.13.2'
+    implementation 'org.apache.logging.log4j:log4j-slf4j-impl:2.13.2'
+    implementation 'org.apache.logging.log4j:log4j-jcl:2.13.2'
+    implementation 'com.beust:jcommander:1.78'
+    implementation 'org.yaml:snakeyaml:1.24'
+    implementation 'commons-io:commons-io:2.6'
+    implementation 'org.apache.commons:commons-compress:1.20'
+    implementation 'org.apache.commons:commons-lang3:3.9'
+    implementation 'commons-beanutils:commons-beanutils:1.9.4'
+    implementation 'com.github.oshi:oshi-json:3.13.4'
+    implementation 'com.jcraft:jsch:0.1.55'
+    implementation 'com.vdurmont:semver4j:3.1.0'
+    implementation 'org.beryx:text-io:3.4.0'
+    implementation 'jline:jline:2.14.6'
+    implementation 'javax.xml.bind:jaxb-api:2.4.0-b180830.0359'
+    implementation 'com.sun.xml.bind:jaxb-core:2.3.0.1'
+    implementation 'com.sun.xml.bind:jaxb-impl:2.3.2'
+    testImplementation 'org.junit.platform:junit-platform-launcher:1.5.2'
+    testImplementation 'org.junit.jupiter:junit-jupiter-engine:5.5.2'
+    testImplementation 'org.junit.vintage:junit-vintage-engine:5.5.2'
+    testImplementation 'org.junit.platform:junit-platform-surefire-provider:1.3.2'
+    testImplementation 'org.mock-server:mockserver-netty:5.5.1'
```

Let's update the test dependencies:

```suggestion

    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.7.2'
    testImplementation 'org.mock-server:mockserver-netty:5.11.2'

    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.7.2'
```

Note: This may require fixing some imports (and static imports) in the Test directory. Merge or rebase origin/main to remove some empty tests. `TestRestConfigFileValidity.java` needs updated:

```diff
-import static org.junit.Assert.*;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```

---

acknowledged, added the above mentioned dependencies and corrected the assertions from `TestRestConfigFileValidity.java` and `TestRestExecCalls.java`